cmd/root.go

```diff
@@ -1,61 +1,491 @@
+		return errors.New("Could not find ID of 'GitHub Number' custom field. Check that it is named correctly.")
+	} else if ghLabelsFieldID == "" {
+		return errors.New("Could not find ID of 'Github Labels' custom field. Check that it is named correctly.")
+	} else if ghStatusFieldID == "" {
+		return errors.New("Could not find ID of 'Github Status' custom field. Check that it is named correctly.")
+	} else if ghReporterFieldID == "" {
+		return errors.New("Could not find ID of 'Github Reporter' custom field. Check that it is named correctly.")
+	} else if isLastUpdateFieldID == "" {
+		return errors.New("Could not find ID of 'Last Issue-Sync Update' custom field. Check that it is named correctly.")
+	}
+
+	log.Debug("All fields have been checked.")
+
+	return nil
+}
+
+func compareIssues(ghClient github.Client, jiraClient jira.Client) error {
+	log.Debug("Collecting issues")
+	ctx := context.Background()
+
+	repo := strings.Split(rootCmdCfg.GetString("repo-name"), "/")
+
+	ghIssues, _, err := ghClient.Issues.ListByRepo(ctx, repo[0], repo[1], &github.IssueListByRepoOptions{
+		Since: since,
+		State: "all",
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if len(ghIssues) == 0 {
+		log.Info("There are no GitHub issues; exiting")
+		return nil
+	}
+	log.Debug("Collected all GitHub issues")
+
+	ids := make([]string, len(ghIssues))
+	for i, v := range ghIssues {
+		ids[i] = fmt.Sprint(*v.ID)
+	}
+
+	jql := fmt.Sprintf("project=%s AND cf[%s] in (%s)",
+		rootCmdCfg.GetString("jira-project"), ghIDFieldID, strings.Join(ids, ","))
+
+	jiraIssues, _, err := jiraClient.Issue.Search(jql, nil)
+	if err != nil {
+		return err
+	}
+
+	log.Debug("Collected all JIRA issues")
+
+	for _, ghIssue := range ghIssues {
+		if ghIssue.PullRequestLinks != nil {
+			continue // This is a PR, not an issue
+		}
+		found := false
+		for _, jIssue := range jiraIssues {
+			id, _ := jIssue.Fields.Unknowns.Int(fmt.Sprintf("customfield_%s", ghIDFieldID))
+			if int64(*ghIssue.ID) == id {
+				found = true
+				if err := updateIssue(*ghIssue, jIssue, jiraClient); err != nil {
+					log.Errorf("Error updating issue %s. Error: %s", jIssue.Key, err)
+				}
+				break
+			}
+		}
+		if !found {
+			if err := createIssue(*ghIssue, jiraClient); err != nil {
+				log.Errorf("Error creating issue for #%d. Error: %s", *ghIssue.Number, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func updateIssue(ghIssue github.Issue, jIssue jira.Issue, client jira.Client) error {
+	log.Debugf("Updating JIRA %s with GitHub #%d", jIssue.Key, *ghIssue.Number)
+
+	anyDifferent := false
+
+	fields := jira.IssueFields{}
+	fields.Unknowns = map[string]interface{}{}
+
+	if *ghIssue.Title != jIssue.Fields.Summary {
+		anyDifferent = true
+		fields.Summary = *ghIssue.Title
+	}
+
+	if *ghIssue.Body != jIssue.Fields.Description {
+		anyDifferent = true
+		fields.Description = *ghIssue.Body
+	}
+
+	key := fmt.Sprintf("customfield_%s", ghStatusFieldID)
```

Comment:
you should create a constant for the `customFieldTemplateString = "customefield_%s"` to avoid errors going forward since this is used to heavily.